src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -119,13 +128,58 @@ def serialize(self, result, root, snuba_event, event_id=None):
 
             # For the light response, the parent will be unknown unless it is a direct descendent of the root
             trace_results.append(
-                serialize_event(snuba_event, root["id"] if root_span is not None else None)
+                self.serialize_event(
+                    snuba_event, root["id"] if root_span is not None else None, None
+                )
             )
 
         event = eventstore.get_event_by_id(snuba_event["project_id"], event_id)
         for span in event.data.get("spans", []):
             if span["span_id"] in parent_map:
                 child_event = parent_map[span["span_id"]]
-                trace_results.append(serialize_event(child_event, event_id))
+                trace_results.append(self.serialize_event(child_event, event_id, None))
 
         return trace_results
+
+
+class OrganizationEventsTraceEndpoint(OrganizationEventsTraceEndpointBase):
+    def serialize_event(self, *args, **kwargs):
+        event = super().serialize_event(*args, **kwargs)
+        event["children"] = []
+        return event
+
+    def serialize(self, parent_map, root, warning_extra, snuba_event=None, event_id=None):
+        """ For the full event trace, we return the results as a graph instead of a flattened list """
+        parent_events = {}
+        parent_events[root["id"]] = result = self.serialize_event(root, None, 0, True)
+
+        current_event = root
+        to_check = deque([root])
+        iteration = 0
+        while to_check:
+            current_event = to_check.popleft()
+            event = eventstore.get_event_by_id(current_event["project_id"], current_event["id"])
+            previous_event = parent_events[current_event["id"]]
+            for child in [
+                item for item in event.data.get("spans", []) if item["span_id"] in parent_map
+            ]:
+                # Avoid potential span loops by popping, so we don't traverse the same nodes twice
```

Comment:
Would it be a good idea to log this happening so we are aware of whether or not it's even possible?

Reply:
Hrmm so i don't think the log would be very actionable (we're already handling the case, and knowing it happens won't change this approach imo)
And its also adding complexity cause I'd have to track visited nodes in a new variable instead.

Open to adding it if there's a way to action the log that I'm not thinking of though :+1:

Reply:
Fair point. If it does happen, there really isn't much we can do here. No problems with not logging this,